src/Nancy/Json/Simple/NancySerializationStrategy.cs

```diff
@@ -0,0 +1,231 @@
+                    ? Nullable.GetUnderlyingType(type) 
+                    : type;
+
+                return value == null
+                    ? null
+                    : Enum.Parse(typeToParse, value.ToString(), true);
+            }
+
+            var primitiveConverter = this.FindPrimitiveConverter(type);
+            if (primitiveConverter != null)
+            {
+                return primitiveConverter.Deserialize(value, type);
+            }
+
+            var valueDictionary = value as IDictionary<string, object>;
+            if (valueDictionary == null)
+            {
+                return base.DeserializeObject(value, type);
+            }
+
+            var javascriptConverter = this.FindJavaScriptConverter(type);
+            if (javascriptConverter != null)
+            {
+                return javascriptConverter.Deserialize(valueDictionary, type);
+            }
+
+            if (!type.IsGenericType)
+            {
+                return base.DeserializeObject(value, type);
+            }
+
+            var genericType = type.GetGenericTypeDefinition();
+            var genericTypeConverter = this.FindJavaScriptConverter(genericType);
+
+            if (genericTypeConverter == null)
+            {
+                return base.DeserializeObject(value, type);
+            }
+
+            var values = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+            var genericArguments = type.GetGenericArguments();
+
+            for (var i = 0; i < genericArguments.Length; i++)
+            {
+                var deserializedObject = this.DeserializeObject(valueDictionary.Values.ElementAt(i),
+                    genericArguments[i]);
+
+                values.Add(valueDictionary.Keys.ElementAt(i), deserializedObject);
+            }
+
+            return genericTypeConverter.Deserialize(values, type);
+        }
+
+        /// <summary>
+        /// Serialize an object
+        /// </summary>
+        /// <param name="input">The object to serialize</param>
+        /// <param name="output">The serialized object</param>
+        /// <returns></returns>
+        protected override bool TrySerializeKnownTypes(object input, out object output)
+        {
+            var dynamicValue = input as DynamicDictionaryValue;
+            if (!ReferenceEquals(dynamicValue, null) && dynamicValue.HasValue)
+            {
+                output = dynamicValue.Value;
+                return true;
+            }
+
+            var inputType = input.GetType().GetTypeInfo();
+            if (this.TrySerializeJavaScriptConverter(input, out output, inputType))
+            {
+                return true;
+            }
+
+            if (this.TrySerializePrimitiveConverter(input, ref output, inputType))
+            {
+                return true;
+            }
+
+            var type = input as Type;
+            if (type != null)
+            {
+                output = type.GetTypeInfo().FullName;

```

Comment:
This case handles the instance where the input we wish to serialize is an instance of System.Type. Changing this to inputType.FullName causes the following test to fail: https://github.com/NancyFx/Nancy/blob/master/test/Nancy.Tests/Unit/JsonFormatterExtensionsFixtures.cs#L68